ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorDemoController.swift

```diff
@@ -146,30 +141,28 @@ class ColorDemoController: UIViewController {
 
     @objc private func segmentedControlValueChanged(sender: Any) {
         if let segmentedControl = sender as? SegmentedControl {
-            let windowType = colorProviderThemes[segmentedControl.selectedSegmentIndex].demoColorTheme.windowType
-            let colorThemeHost = view.window?.windowScene?.delegate as? ColorThemeHosting
-            ColorDemoController.themeWindowType = windowType
-
-            if let navigationController = navigationController {
-                navigationController.popViewController(animated: false)
-                colorThemeHost?.updateToWindowWith(type: windowType, pushing: self)
+            let selectedSegmentIndex = segmentedControl.selectedSegmentIndex
+            let provider = colorProviderThemes[selectedSegmentIndex].demoColorTheme.provider
+            let window = self.view.window
+            ColorDemoController.providerType = colorProviderThemes[selectedSegmentIndex].demoColorTheme.providerType
+
+            if let provider = provider as? ColorProviding {
+                DemoListViewController.updateColorProviderFor(window: window!, provider: provider)
+            } else {
+                // If provider doesn't conform to ColorProviding, fallback on default theme
+                DemoListViewController.updateColorProviderFor(window: window!, provider: DemoColorDefaultTheme() as ColorProviding)
             }
 
-            let window = colorProviderThemes[segmentedControl.selectedSegmentIndex].demoColorTheme.window
-            if let colorProvider = window as? ColorProviding {
-                Colors.setProvider(provider: colorProvider, for: window)
-            }
+            tableView.reloadData()
+            segmentedControl.didMoveToWindow()
```

Comment:
why do you need didmovetowindow?

Reply:
instead of the OS api, calling updateWindowSpecificColors would be more appropriate.